### PR TITLE
Cmake migration

### PR DESCRIPTION
**Pull request by sersajur**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -21,9 +21,21 @@ moc_*.cpp
...
+
+*.vscode
+
+# Mac
+
+.DS_Store
+
+# Personal need
+
+backlog
```
Comment (Youw) [inline .gitignore]: no newline at the end of file
here and elsewhere

**Comment by sersajur** [comment]
Connected to #10

File: .gitignore (hunk 1 above)
Comment (sersajur) [reply .gitignore]: Added with
https://github.com/sersajur/QPentago/pull/11/commits/15b4eb91f09dbc6bdf4bfd9ac84c80d8c2a4be72

File: external/gtest/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# We need thread support
+find_package(Threads REQUIRED)
+
+# Enable ExternalProject CMake module
+include(ExternalProject)
+
+# Download and install GoogleTest
+ExternalProject_Add(
```
Comment (Youw) [inline external/gtest/CMakeLists.txt]: Think about [git submodule](https://git-scm.com/book/en/v2/Git-Tools-Submodules) - a "git way" to add external dependencies.

**Comment by Youw** [comment]
Looks good. Time to merge.

**Pull request merged**